crates/prover-services/sharp-service/tests/lib.rs

```diff
@@ -0,0 +1,154 @@
+use alloy::primitives::B256;
+use cairo_vm::vm::runners::cairo_pie::CairoPie;
+use httpmock::standalone::start_standalone_server;
+use httpmock::MockServer;
+use lazy_static::lazy_static;
+use prover_client_interface::{ProverClient, Task, TaskId, TaskStatus};
+use rstest::rstest;
+use serde_json::json;
+use sharp_service::{split_task_id, SharpProverService};
+use snos::sharp::CairoJobStatus;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Mutex;
+use std::thread::{spawn, JoinHandle};
+use tokio::task::LocalSet;
+use utils::env_utils::get_env_var_or_panic;
+use utils::settings::default::DefaultSettingsProvider;
+
+// To start a standalone server to be used with sharp client mocking
+lazy_static! {
+    static ref STANDALONE_SERVER: Mutex<JoinHandle<Result<(), String>>> = Mutex::new(spawn(|| {
+        let srv = start_standalone_server(5000, false, None, false, usize::MAX, std::future::pending());
+        let runtime = tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap();
+        LocalSet::new().block_on(&runtime, srv)
+    }));
+}
```

Comment:
so we did this thing in settlement client where we added a new `with_test_settings` method where we can supply custom things like the mock server URL. i think that might be a better approach here and would be easier to parallelise later as well, wdyt?